pyspark_huggingface/compat/datasource.py

```diff
@@ -0,0 +1,198 @@
+from typing import TYPE_CHECKING, Iterator, List, Optional, Union
+
+import pyspark
+
+
+if int(pyspark.__version__.split(".")[0]) >= 4 and ("dev0" not in pyspark.__version__ and "dev1" not in pyspark.__version__):
```

Comment:
```suggestion
if int(pyspark.__version__.split(".")[0]) >= 4 and pyspark.__version__ not in ("4.0.0.dev0", "4.0.0.dev1"):
```
we don't want this to match for example `4.1.0.dev0`